### PR TITLE
cargo,clippy,rustc,rustfmt: 1.80.1 -> 1.81.0

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

Fixes: CVE-2024-24576

Built:

- [x] fd.aarch64-darwin
- [x] fd.aarch64-linux
- [x] fd.x86_64-darwin
- [x] fd.x86_64-linux
- [x] firefox.x86_64-linux
- [x] thunderbird.x86_64-linux

**Please help test Darwin.**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/rust/rustc.nix [hunk 1]
```diff
@@ -109,6 +109,9 @@ in stdenv.mkDerivation (finalAttrs: {
     "--tools=rustc,rustdoc,rust-analyzer-proc-macro-srv"
     "--enable-rpath"
     "--enable-vendor"
+    # For Nixpkgs it makes more sense to use stdenv's linker than
+    # letting rustc build its own.
+    "--disable-lld"
```
Comment (winterqt) [inline pkgs/development/compilers/rust/rustc.nix:114]: Maybe this is worth splitting into a different commit/what made you change this with this bump?

Comment (alyssais) [reply pkgs/development/compilers/rust/rustc.nix:114]: It started complaining if I didn't set it.

Comment (alyssais) [reply pkgs/development/compilers/rust/rustc.nix:114]: So it's not a change, it's preserving the previous behaviour.

**Comment by emilazy** [comment]
Successfully built `fd` on `aarch64-darwin`. My local machine is too weird right now to confirm `x86_64-darwin`.

**Comment by emilazy** [comment]
Confirmed `fd` on `x86_64-darwin` too.

**Comment by alyssais** [comment]
This breaks `fastCross` cross compilation.  I'm investigating, but worst case we can just disable that, and cross compiling will mean having to compile a whole new rustc again.

**Comment by alyssais** [comment]
Bisected to https://github.com/rust-lang/rust/pull/127866.

**Comment by alyssais** [comment]
One of rust-lang/rust@2593673435b0f524fd5b09541d4ac611e167205e, rust-lang/rust@c47dc4be4a9495e90b16bbc2c28e89fb216f3fa8 fixes this.  I'll figure out which one tomorrow.

**Comment by alyssais** [comment]
It's the latter, so I guess I'll apply it and we can re-enable fastCross once https://github.com/rust-lang/rust/pull/130034 is merged.

**Comment by chayleaf** [comment]
to clarify for anyone else confused about this, this fixes CVE-2024-43402, not CVE-2024-24576 (which got fixed in 1.77.2). Both are Windows-only, so the impact for Nix is zero, or at least very close to zero.

**Comment by alyssais** [comment]
Nixpkgs supports building Rust programs for Windows.

**Comment by alyssais** [comment]
Sorry for the wrong CVE ID though.

**Pull request merged**